### ISSUE TITLE
Gaufrette storage + Azure Blob Storage is not working

### ISSUE THREAD

**Issue by juulhobert** [dustin10/VichUploaderBundle#708]
I've configured the Vich uploader with the Gaufrette storage. Then I configured Gaufrette to store it's data in the Azure Blob Storage. When you do this and upload a file it will cause the following exception:

```
Value: The specified blob does not exist.
details (if any): ﻿<?xml version="1.0" encoding="utf-8"?><Error><Code>BlobNotFound</Code><Message>The specified blob does not exist.
```
This happens because the GaufretteStorage class writes the metadata before it writes the data. The blob storage does not exists at the time it wants to write the meta data.

**Comment by xixi773** [comment]
I got the same issue, an easy way to solve this is to set metadata after Gaufrette Storage save the data.

You can see it on this repository : https://github.com/tech-ec/VichUploaderBundle/commit/f18539494e5af0090045af88a0d125aaa5175e6b

It will be great if this little change can be add to a next release.

**Comment by ThBM** [comment]
Hi,
Is there any reason this fix is not merged yet?
Without it, Azure Blob Storage cannot be used for now ... too bad.

Thank you in advance.

**Comment by garak** [comment]
Fixed in #1160
